python/mlcroissant/mlcroissant/_src/core/json_ld.py

```diff
@@ -167,13 +169,34 @@ def recursively_populate_jsonld(entry_node: Json, id_to_node: dict[str, Json]) -
     return entry_node
 
 
+def check_valid_ids(data: Json, ctx: Context) -> bool:
+    """Checks that the given json contains valid `@id`s."""
+    if hasattr(data, "items"):
+        for k, v in data.items():
+            if k == "@id" and re.match(_ID_REGEX, v):
+                ctx.issues.add_error(
+                    f"The dataset contains a wrong `@id`: '{v}'. Note that currently we"
+                    " do not support `@id`s containing whitespaces (not even"
+                    " URL-escaped)."
+                )
+                return False
+            if isinstance(v, dict):
+                check_valid_ids(v, ctx)
+            elif isinstance(v, list):
+                for d in v:
+                    check_valid_ids(d, ctx)
+    return True
+
+
 def expand_jsonld(data: Json, ctx: Context) -> Json:
     """Expands a Croissant JSON to a nested JSON-LD with expanded.
 
     For this we use RDFLib. RDFLib expands the CURIE of the form "rdf:type" into their
     full expression, but RDFLib also flattens the JSON-LD in a list of nodes. We then
     need to reconstruct the hierarchy.
     """
+    if not check_valid_ids(data=data, ctx=ctx):
+        ctx.issues.add_error("There are wrong ids in this dataset.")
```

Comment:
This is not necessary as `check_valid_ids` already populate issues?

Reply:
Done.